integrations/terraform-cloud/main.py

```diff
@@ -136,21 +131,20 @@ async def fetch_runs_for_workspace(
             )
         ]
 
-    async def fetch_runs_for_all_workspaces() -> ASYNC_GENERATOR_RESYNC_TYPE:
-        async for workspaces in terraform_client.get_paginated_workspaces():
-            logger.info(
-                f"Received {len(workspaces)} batch workspaces... fetching its associated {kind}"
-            )
+    async for workspaces in terraform_client.get_paginated_workspaces():
+        logger.info(
+            f"Received {len(workspaces)} batch workspaces... fetching its associated {kind}"
+        )
 
-            tasks = [fetch_runs_for_workspace(workspace) for workspace in workspaces]
-            for completed_task in asyncio.as_completed(tasks):
-                workspace_runs = await completed_task
+        for chunk in [
+            workspaces[i : i + CHUNK_SIZE]
+            for i in range(0, len(workspaces), CHUNK_SIZE)
+        ]:
+            chunk_results = await gather(*[fetch_runs_for_workspace(w) for w in chunk])
```

Comment:
Is there a specific reason for replacing as_completed with gather ? waiting for all tasks within a chunk to complete before moving on to the next chunk appears to impede performance in this context. Please correct me.